bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java

```diff
@@ -0,0 +1,949 @@
+        } catch (InterruptedException | IOException e) {
+            String message = e.toString();
+            logger.error("Could not start the communication with the server. {}", message);
+        }
+    }
+
+    /**
+     * Method to check if communication with Qbus Server is active
+     *
+     * @return True if active
+     */
+    public boolean communicationActive() {
+        return qSocket != null;
+    }
+
+    /**
+     * Method to check if communication with Qbus Client is active
+     *
+     * @return True if active
+     */
+
+    public boolean clientConnected() {
+        return ctdConnected;
+    }
+
+    /**
+     * Runnable that handles incomming communication from Qbus server.
+     * <p>
+     * The thread listens to the TCP socket opened at instantiation of the {@link QbusCommunication} class
+     * and interprets all incomming json messages. It triggers state updates for active channels linked to the
+     * Qbus outputs. It is started after initialization of the communication.
+     *
+     * @return
+     * @throws IOException
+     *
+     */
+    private void qEvents() throws IOException {
+        String qMessage;
+
+        listenerStopped = false;
+        qEventsRunning = true;
+
+        BufferedReader reader = this.qIn;
+
+        if (reader == null) {
+            throw new IOException("Bufferreader for incoming messages not initialized.");
+        }
+
+        while (!Thread.currentThread().isInterrupted() && ((qMessage = reader.readLine()) != null)) {
+            readMessage(qMessage);
+        }
+
+        if (!listenerStopped) {
+            qEventsRunning = false;
+
+            QbusBridgeHandler handler = bridgeCallBack;
+
+            if (handler != null) {
+                ctdConnected = false;
+                handler.bridgeOffline("No communication with Qbus server");
+            }
+        }
+
+        qEventsRunning = false;
+        logger.trace("Event listener thread stopped on thread {}", Thread.currentThread().getId());
+    };
+
+    /**
+     * Called by other methods to send json data to Qbus.
+     *
+     * @param qMessage
+     * @throws InterruptedException
+     */
+    synchronized void sendMessage(Object qMessage) throws InterruptedException {
+        PrintWriter writer = qOut;
+        String json = gsonOut.toJson(qMessage);
+        QbusBridgeHandler handler = bridgeCallBack;
+
+        if (writer != null) {
+            writer.println(json);
+            // Delay after sending data to improve scene execution
+            TimeUnit.MILLISECONDS.sleep(250);
+
+        }
+        if ((writer == null) || (writer.checkError())) {
+            logger.warn("Error sending message, trying to restart communication");
+            try {
+                restartCommunication();
+            } catch (InterruptedException e) {
+                if (handler != null) {
+                    handler.bridgeOffline("No communication with Qbus server. " + e.toString());
+                }
```

Comment:
As `sendMessage()` already throws the `InterruptedException`, you could simply remove this catch clause.